example/lib/main.dart

```diff
@@ -59,22 +75,85 @@ class _MyAppState extends State<MyApp> {
     }
   }
 
+  void applyDarkTheme() async {
+    try {
+      final ByteData bytes =
+          await DefaultAssetBundle.of(context).load(_documentPath);
+      final Uint8List list = bytes.buffer.asUint8List();
+
+      final tempDir = await getTemporaryDirectory();
+      final tempDocumentPath = '${tempDir.path}/$_documentPath';
+
+      final file = await File(tempDocumentPath).create(recursive: true);
+      file.writeAsBytesSync(list);
+
+      Pspdfkit.present(tempDocumentPath, {
```

Comment:
Are there actually any reasons this class is not called `PSPDFKit`? (casing) I'd vote for rebranding it unless there's some Flutter/Dart specific reason not to.

Reply:
I know having it lowercase is a bit annoying but unfortunately the name `PSPDFKit` as a plugin would be against the naming convention adopted by Flutter. I don't know how bad it would be but the wizard, for instance, does not allow me to create a `PSPDFKit` plugin, check the screenshot. We can opt to go for a manual rename and rebranding it, but it would be a dagnerous way.
(I would not do that)

![Screenshot 2019-03-18 18 31 19](https://user-images.githubusercontent.com/3308503/54550419-4443b000-49ac-11e9-8d0d-306737e4629f.png)

Reply:
But doesn't it only warn about the package name though? Capitalization rules seem to be the same we use on Java too: https://www.dartlang.org/guides/language/effective-dart/style#identifiers

That is:
* Upper camel case for type names.
* No capitalization of acronyms longer than 2 characters.
* ...

Since we follow the same naming rules for Java and Kotlin (and also mostly all other platforms, if you ignore the iOS type prefixes) we are in the same category of naming here too. Having that said, I'd say it makes sense for consistency to use the same "brand aware" capitalization rule here too.

Reply:
No, it does not warn because of the package name, (that is in the next screen), it warns about the plugin name.

The rules are quite strict, also because there are several steps involved here, let shed some lights, I'll describe part of the class generation:

- When the plugin is chosen it must start with a capital letter (`pspdfkit`)
- A method channel calling the exact same name is generated: 
`static const MethodChannel _channel = const MethodChannel('pspdfkit');`
- A class with the first letter capitalized is generated (then `Pspdfkit`)
the method channel is placed inside this class among other things
- There are also other exceptions that I'll not go to explain into detail (but I will if you want to) like the underscores `_` underscores are allowed as word separators, they will be used for the channel method and removed for the class generation in favor of the upper camel case notation.
- The plugin name will be also used in the yaml file